test/api/test_tools.py

```diff
@@ -1407,6 +1416,19 @@ def _run(self, tool_id, history_id, inputs, assert_ok=False, tool_version=None):
         else:
             return create_response
 
+    def _upload_from_url(self, url):
```

Comment:
In this same directory there is a test_tools_upload.py test case filled with tons of upload tests - can you move these there?

Reply:
Hmm, that's only in dev. Should I create the file ?

Reply:
Duh - good point. Lets keep it as is - I'll try to remember to move things around after we merge this forward.